### PR TITLE
Create application skeleton

### LINKED ISSUE #1: Create basic application layout & functionality
- [x] Core infrastructure
- [x] Board layout
- [x] Create, name, rename board
- [x] Add/remove/move cards to/from columns

### PR DESCRIPTION
**Pull request by Trinityyi**
Resolves #1 
Related: #36

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -7,37 +7,14 @@
     <meta name="theme-color" content="#000000" />
     <meta
       name="description"
-      content="Web site created using create-react-app"
+      content="Manage your productivity"
```
Comment (Chalarangelo) [inline public/index.html:10]: [TODO] We should add additional metadata to the page, as necessary.

Some related material: https://ahrefs.com/blog/seo-meta-tags/

File: src/components/App.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+import { createStore } from 'redux';
+import MainArea from './MainArea';
+import Header from './Header';
+import store from '../store';
+
+
```
Comment (Chalarangelo) [inline src/components/App.jsx:8]: [formatting] :scissors: the extra empty line.

File: src/components/Column.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Column = ({
+  cards,
+  title
+}) => {
+
```
Comment (Chalarangelo) [inline src/components/Column.jsx:8]: [formatting] :scissors: the extra empty line. Also consider changing to an implicit return, as there are no other statements as of yet.

File: src/components/Column.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        {cards.map(card => (
+          <li key={card.id}>{card.title}</li>
+        ))}
+      </ul>
+    </div>
+  );
+};
+
+Column.propTypes = {
+  cards: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
```
Comment (Chalarangelo) [inline src/components/Column.jsx:22]: [proptypes] Let's describe the `shape` here when we decide on it.

File: src/components/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Header = () => {
+
```
Comment (Chalarangelo) [inline src/components/Header.jsx:5]: [formatting] :scissors: the extra empty line. Consider converting to implicit return as well.

File: src/components/MainArea.jsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  );
+};
+
+MainArea.propTypes = {
+  name: PropTypes.string.isRequired,
+  setBoardName: PropTypes.func.isRequired
+};
+
+const mapStateToProps = state => {
+  return {name: state.name};
```
Comment (Chalarangelo) [inline src/components/MainArea.jsx:56]: [formatting] Pad the object with whitespaces like `{ name: state.name }` for consistency with the rest of the code.

File: src/hooks/useClickOutside.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react';
+
+// Reference: https://www.30secondsofcode.org/react/s/use-click-outside
+const useClickOutside = (ref, callback) => {
```
Comment (Chalarangelo) [inline src/hooks/useClickOutside.js:4]: [docs] We should add a JSDoc description to this.

File: src/styles/_reset.scss [hunk 8]
```diff
@@ -0,0 +1,211 @@
+/*! normalize.css v8.0.1 | MIT License | github.com/necolas/normalize.css */
```
Comment (Chalarangelo) [inline src/styles/_reset.scss:1]: [TODO] I should clean up this file and add customized resets as necessary.

File: src/styles/_reset.scss [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+   ========================================================================== */
+
+/**
+ * 1. Change the font styles in all browsers.
+ * 2. Remove the margin in Firefox and Safari.
+ */
+
+button,
+input,
+optgroup,
```
Comment (Chalarangelo) [inline src/styles/_reset.scss:122]: [TODO] I should :scissors: this, probably will never be used in our codebase.

File: src/components/Column.jsx (hunk 3 above)
Comment (Trinityyi) [reply src/components/Column.jsx]: Fixed in 9fc0eb3

File: src/components/Column.jsx (hunk 4 above)
Comment (Trinityyi) [reply src/components/Column.jsx]: 9fc0eb3

File: src/store/index.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+  switch (action.type) {
+  case SET_BOARD_TITLE:
+    return {
+      ...state,
+      board: {
+        ...state.board,
+        title: action.title
+      }
+    };
+
```
Comment (Chalarangelo) [inline src/store/index.js:26]: :scissors: the empty line

File: src/store/index.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+  name: 'Board',
+  board: {
+    id: 0,
+    title: 'Board',
+    columnIds: []
+  },
+  columns: {
+  },
+  cards: {
+  }
```
Comment (Chalarangelo) [inline src/store/index.js:13]:
```suggestion
  columns: {},
  cards: {}
```

File: src/store/index.js [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+        [action.columnId]: {
+          ...state.columns[action.columnId],
+          cardIds: [
+            ...state.columns[action.columnId].cardIds,
+            action.cardId
+          ]
+        }
+      }
+    };
+
```
Comment (Chalarangelo) [inline src/store/index.js:57]: :scissors: the empty line

File: src/styles/_main_area.scss [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+.main-area-content {
+  display: block;
+  top: 0;
+  left: 0;
+  height: 100%;
+  width: 100%;
+  background: #A8C6FA;
+  margin:0;
+  display: flex;
+  align-items: start;
```
Comment (Chalarangelo) [inline src/styles/_main_area.scss:17]: :sweat_smile: This should be `flex-start`.

**Pull request merged**